### PR TITLE
Update changelog to 0.7

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
This updates the changelog to include all changes up to commit ca6dbf6. I would like a couple of reviewers to ensure I didn't forget anything.

Updates #1217.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,64 @@
-# 0.7.0 alpha.1
-
-Currently the changelog for v0.7.0 is incomplete, this tracked in issue #1217.
+# 0.7.0
+
+Version 0.7 of Mio contains various major changes compared to version 0.6.
+Overall a large number of API changes have been made to reduce the complexity of
+the implementation and remove overhead where possible.
+
+Also the upgrade guide in docs/upgrade_guide.md.
```
Comment (kleimkuhler) [inline CHANGELOG.md]: Maybe change this to something like: `Please refer to the upgrade guide in docs/upgrade_guide.md for additional information`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
+
+## Added
+
+* New `Interest` structure that replaces `Ready` in registering event sources.
```
Comment (kleimkuhler) [inline CHANGELOG.md]: `New` is a little redundant here as the section is titled `Added`. Maybe change this to: `'Interest' structure that ..`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
+* `Registry` structure was added to separate registering and polling.
```
Comment (kleimkuhler) [inline CHANGELOG.md]: How about: `'Registry' structure that separates the registering and polling functionality.`

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,10 +1,64 @@
...
+* `Waker` was added to allow another thread to wake a thread polling `Poll`.
```
Comment (kleimkuhler) [inline CHANGELOG.md]: `'Waker' structure that allows another ..`?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,10 +1,64 @@
...
+* Added Unix Domain Socket (UDS) types: `UnixDatagram`, `UnixListener` and
```
Comment (kleimkuhler) [inline CHANGELOG.md]: Similar to above, `Added` is a little redundant here.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,10 +1,64 @@
...
+  `UnixStream`.
+
+## Removed
+
+* All code deprecated in 0.6 was removed in 0.7.
+* Support for Fuchsia was removed, the code was unmaintained.
```
Comment (kleimkuhler) [inline CHANGELOG.md]: How about: `.. was removed as the code was unmaintained.`

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,10 +1,64 @@
...
+* Support for Bitrig was removed, rustc dropped support for it also.
+* `UnixReady` was merged into `Ready`.
+* Custom user-space readiness queue was removed, this includes the public
+  `Registration` and `SetReadiness` types.
+* `PollOpt` was removed, now all registrations use edge-triggers. See the
```
Comment (kleimkuhler) [inline CHANGELOG.md]: thoughts on: `was removed and all registrations now use edge-triggers.`

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,10 +1,64 @@
...
+  upgrade guide on how to process event using edge-triggers.
+* The network types (types in the `net` module) now support only the same API as
+  found in the standard library, various methods on the types were removed.
+* `TcpStream` now supports vectored I/O.
+* `Poll::poll_interruptible` was removed and instead `Poll::poll` doesn't hide
```
Comment (kleimkuhler) [inline CHANGELOG.md]: This is a little confusing to read. How about: `Poll::poll_interruptible is no longer provided. Poll::poll will now return an error if one occurs.`

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,10 +1,64 @@
...
+  `Poll::poll_interruptible` renamed to `Poll::poll`.
+* `From<usize>` is removed from `Token`, the internal field is still public, so
+  `Token(my_token)` can still be used.
+
+## Changed
 
 * The `join_multicast_v4` and `leave_multicast_v4` methods now take their
   `Ipv4Addr` arguments by value rather than by reference.
-* Fix lazycell related compilation issues.
+* Various documentation improvements were made  around correct usage of `Poll`
```
Comment (kleimkuhler) [inline CHANGELOG.md]: Extra space in `.. made  around ..`

**Comment by Thomasdezeeuw** [comment]
@kleimkuhler I've addressed all your feedback.

**Pull request merged**